hilti/toolchain/src/compiler/codegen/types.cc

```diff
@@ -1111,6 +1111,26 @@ const CxxTypeInfo& CodeGen::_getOrCreateTypeInfo(const hilti::Type& t) {
         });
 }
 
+cxx::Expression CodeGen::_applyLhs(cxx::Expression expr, const Type& type, bool lhs) {
+    if ( ! lhs || expr.isLhs() )
+        return expr;
+
+    auto tmp = addTmp("lhs", compile(type, TypeUsage::Storage));
+    cxx::Expression result;
+
+    if ( type.isA<type::ValueReference>() )
+        result = cxx::Expression{fmt("(%s=(%s).asSharedPtr())", tmp, expr), cxx::Side::LHS}; // avoid copy
+    else
+        result = cxx::Expression{fmt("(%s=(%s))", tmp, expr), cxx::Side::LHS};
+
+#if 0
+    // This can help show where LHS conversions happen unexpectedly (they should be very rare).
+    std::cerr << "LHS| " << type.typename_() << ": " << expr << " -> " << result << std::endl;
+#endif
```

Comment:
Does it make sense to add this to a debug stream?

Reply:
We don't have a `codegen` debug stream yet, but I'll create one, makes sense.